### PR TITLE
Bug fix: pymongo 3.0 defines 'version' instead of '__version__'

### LINKED ISSUE #619: mongo-connector 2.5.0 no longer works with pymongo 3.0
mongo-connector now assumes pymongo.__version__ exists but it doesn't in pymongo 3.0

```sh-session
$ virtualenv env
$ env/bin/pip install mongo-connector==2.5.0 pymongo==3.0
$ env/bin/mongo-connector
Logging to /tmp/mongo-connector.log.
Traceback (most recent call last):
  File "env/bin/mongo-connector", line 9, in <module>
    load_entry_point('mongo-connector==2.5.0', 'console_scripts', 'mongo-connector')()
  File "/tmp/env/local/lib/python2.7/site-packages/mongo_connector/util.py", line 94, in wrapped
    func(*args, **kwargs)
  File "/tmp/env/local/lib/python2.7/site-packages/mongo_connector/connector.py", line 1209, in main
    log_startup_info()
  File "/tmp/env/local/lib/python2.7/site-packages/mongo_connector/connector.py", line 1192, in log_startup_info
    LOG.always('pymongo version: %s', pymongo.__version__)
AttributeError: 'module' object has no attribute '__version__'
```

### PR DESCRIPTION
**Pull request by ShaneHarvey**
Fixes #619.

### CODE REVIEW

**Comment by ShaneHarvey** [comment]
Running the test suite with PyMongo 3.0: https://travis-ci.org/ShaneHarvey/mongo-connector/builds/200552760

File: mongo_connector/connector.py [hunk 1]
```diff
@@ -1189,7 +1189,12 @@ def log_startup_info():
                     __version__)
     LOG.always('Python version: %s', sys.version)
     LOG.always('Platform: %s', platform.platform())
-    LOG.always('pymongo version: %s', pymongo.__version__)
+    pymongo_version = 'unknown'
```
Comment (behackett) [inline mongo_connector/connector.py]: 'version' has existed for forever (and still exists today), so this all can be simplified to:

```
if hasattr(pymongo, '__version__'):
    pymongo_version = pymongo.__version__
else:
    pymongo_version = pymongo.version
```

You could also just use 'version', but I guess checking for __version__ is a bit more future proof.

Comment (ShaneHarvey) [inline mongo_connector/connector.py]: Simplified.

**Comment by behackett** [comment]
LGTM

**Pull request merged**